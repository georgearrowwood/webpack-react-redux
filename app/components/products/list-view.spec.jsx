import React from 'react';
import { mount } from 'enzyme';

import ProductsList from './list-view';

const mockProductsList = [
  { id: 1, title: 'test product 1' },
  { id: 2, title: 'test product 2' },
];

let wrapper;
const deleteClickHandlerMock = jest.fn();

beforeAll(() => {
  wrapper = mount(<ProductsList
    products={mockProductsList}
    deleteHandler={deleteClickHandlerMock}
  />);
});

describe('ProductsList View', () => {
  it('Renders products component with properties', () => {
    expect(wrapper).toMatchSnapshot();
    expect(Array.isArray(wrapper.prop('products'))).toEqual(true);
    expect(wrapper.prop('products').length).toEqual(2);
    expect(typeof wrapper.prop('deleteHandler')).toEqual('function');
  });

  it('Renders products list items and its titles', () => {
    expect(wrapper.find('ul').find('li').length).toEqual(2);
    const firstProduct = wrapper.find('ul').find('li').at(0);
    expect(firstProduct.find('span').text()).toEqual(`${mockProductsList[0].id} - ${mockProductsList[0].title} :`);
  });

  it('Rendered products delete buttons rendered and work', () => {
    const firstProduct = wrapper.find('ul').find('li').at(0);
    expect(firstProduct.find('a').text()).toEqual('X');
    firstProduct.find('a').simulate('click');
    expect(deleteClickHandlerMock.mock.calls.length).toBe(1);
    expect(deleteClickHandlerMock.mock.calls[0][0]).toBe(1);
    const secondProduct = wrapper.find('ul').find('li').at(1);
    secondProduct.find('a').simulate('click');
    expect(deleteClickHandlerMock.mock.calls[1][0]).toBe(2);
  });
});
