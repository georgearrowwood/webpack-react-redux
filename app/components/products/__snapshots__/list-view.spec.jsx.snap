// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList View Renders products component with properties 1`] = `
<ProductsListView
  deleteHandler={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "title": "test product 1",
      },
      Object {
        "id": 2,
        "title": "test product 2",
      },
    ]
  }
>
  <ul>
    <li
      key="1"
    >
      <span>
        1
         - 
        test product 1
         :
      </span>
      <a
        href="delete/{product.id}"
        onClick={[Function]}
      >
        X
      </a>
    </li>
    <li
      key="2"
    >
      <span>
        2
         - 
        test product 2
         :
      </span>
      <a
        href="delete/{product.id}"
        onClick={[Function]}
      >
        X
      </a>
    </li>
  </ul>
</ProductsListView>
`;
